src/native/s3_client.c

```diff
@@ -998,6 +1019,17 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientMake
         AWS_OP_SUCCESS == aws_apply_java_http_request_changes_to_native_request(
                               env, jni_marshalled_message_data, jni_http_request_body_stream, request_message));
 
+    if (jni_operation_name) {
+        operation_name = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_operation_name);
+        if (operation_name.ptr == NULL) {
+            goto done;
+        }
+        if (operation_name.len == 0) {
+            aws_jni_throw_illegal_argument_exception(env, "Operation name cannot be empty");
+            goto done;
+        }
```

Comment:
Why are we not throwing exception in the first if? Should we just combine these if statements?

Reply:
If `operation_name.ptr == NULL` then the `aws_jni_string_from_string()` call on the previous line failed, and it already raised an exception.

https://github.com/awslabs/aws-crt-java/blob/d3ab7768c426cda20f8ed33695f019e6d2af3bcc/src/native/crt.h#L162-L165